### DISCUSSION TITLE
Allow YYMMDD date pattern

### DISCUSSION THREAD

**Discussion by nod5** [jotaen/xit#18] [File Format]
> The date pattern MUST be either of:
>     `yyyy-mm-dd` or `yyyy/mm/dd` to reference a calendar day
>     `yyyy-mm` or `yyyy/mm` to reference a month period
>     `yyyy` to reference a year period
>     `yyyy-Www` or `yyyy/Www` to reference a week period
>     `yyyy-Qq` or `yyyy/Qq` to reference a quarter period
> 

Allow the `YYMMDD` date pattern (and `YY-Www` and `YYQq`)

**Reasons:**
+ Follows the same larger-to-smaller time units rule.
+ Shorter (type faster, see less clutter).
+ Avoids special characters `-` or `/` which require modifier keys to type on some keyboard layouts.

**Objection 1:**  Some may confuse `YYMMDD` with `DDMMYY`.

**Reply:**  Not if they read the spec (once `YYMMDD` is added). The horrible fact that `DDMMYY` exists in the wild should not force [x]it! into verbosity. A single person user choosing `YYMMDD` is unlikely to get confused by it. In multi user cases the users are still free to adopt a convention to only use the verbose `YYYY` patterns.

**Objection 2:** `YYMMDD` may cause future issues when year 2100 is closer.

**Reply:** Not a practical problem for a long time and the required `->` prefix should make batch converting old [x]it! data from `YYMMDD` to `YYYY` variants easy, if that need arises.

**Comment by jotaen** [comment]
Cool, thanks for that thorough discussion!

This is actually something that I also thought about, because while e.g. `YYYY-MM-DD` is clear and intuitive to read, it is also a bit bulky and might be tiring to type.

I think the date format is a tricky decision, because there are so many variations and conventions around the world. On the one hand, it would be terrific to support everyone’s local style (think `-> 5th May`, but then also: `-> 二零二一年十二月二十五号`), but on the other hand the rules must be both machine-readable and also easy to memorise.

I’m wondering whether this whole topic could be alleviated through good tooling, such as an editor plugin. For example, the [one for Sublime Text](https://packagecontrol.io/packages/xit%21) has smart-completions: (video recorded at 2022-04-02)

https://user-images.githubusercontent.com/3618384/161374754-0980c048-c1a4-471c-aad4-44f5aab956da.mov

**Comment by nod5** [comment]
> there are so many variations and conventions around the world

True, but `5th may` is much more dissimilar from the patterns [x]it! currently accepts and breaks the larger-to-smaller time units rule. `YYMMDD` is _the_ shorthand for the current patterns.

> could be alleviated through good tooling

That speeds up input nicely, but still takes up more visual space than `YYMMDD`.
